custom_components/babybuddy/sensor.py

```diff
@@ -1,409 +1,355 @@
-            ATTR_METHOD: call.data.get(ATTR_METHOD).lower(),
-            ATTR_AMOUNT: call.data.get(ATTR_AMOUNT),
-            ATTR_NOTES: call.data.get(ATTR_NOTES),
+            ATTR_CHILD: self.child[ATTR_ID],
+            ATTR_TEMPERATURE: temperature,
+            ATTR_TIME: date_time,
         }
-
-        baby_buddy.entities_add(endpoint, data)
-
-    def services_notes_add(call):
-        endpoint = ATTR_NOTES
+        if notes:
+            data[ATTR_NOTES] = notes
+
+        await self.coordinator.client.async_post(ATTR_TEMPERATURE, data)
+        await self.coordinator.async_request_refresh()
+
+    async def async_add_weight(
+        self, weight: float, date: date | None = None, notes: str | None = None
+    ) -> None:
+        """Add weight entry."""
+        if not isinstance(self, BabyBuddyChildSensor):
+            _LOGGER.debug("Babybuddy child sensor should be selected")
+            return
         data = {
-            ATTR_CHILD: hass.states.get(call.data.get(ATTR_CHILD)).attributes.get(
-                ATTR_ID
-            ),
-            ATTR_NOTE: call.data.get(ATTR_NOTE),
-            ATTR_TIME: call.data.get(ATTR_TIME),
+            ATTR_CHILD: self.child[ATTR_ID],
+            ATTR_WEIGHT: weight,
+            ATTR_DATE: date or dt_util.now().date(),
         }
+        if notes:
+            data[ATTR_NOTES] = notes
+
+        await self.coordinator.client.async_post(ATTR_WEIGHT, data)
+        await self.coordinator.async_request_refresh()
+
+    async def async_add_note(
+        self, note: str, time: datetime | time | None = None
+    ) -> None:
+        """Add note entry."""
+        if not isinstance(self, BabyBuddyChildSensor):
+            _LOGGER.debug("Babybuddy child sensor should be selected")
+            return
+        try:
+            date_time = get_datetime_from_time(time or dt_util.now())
+        except ValidationError as err:
+            _LOGGER.error(err)
+            return
+        data = {ATTR_CHILD: self.child[ATTR_ID], ATTR_NOTE: note, ATTR_TIME: date_time}
 
-        baby_buddy.entities_add(endpoint, data)
+        await self.coordinator.client.async_post(ATTR_NOTES, data)
+        await self.coordinator.async_request_refresh()
 
-    def services_sleep_add(call):
-        endpoint = ATTR_SLEEP
-        data = {
-            ATTR_CHILD: hass.states.get(call.data.get(ATTR_CHILD)).attributes.get(
-                ATTR_ID
-            ),
-            ATTR_START: call.data.get(ATTR_START),
-            ATTR_END: call.data.get(ATTR_END),
-            ATTR_NOTES: call.data.get(ATTR_NOTES),
-        }
+    async def async_delete_last_entry(self) -> None:
+        """Delete last data entry."""
+        if not isinstance(self, BabyBuddyChildDataSensor):
+            _LOGGER.debug("Babybuddy child data sensor should be selected")
+            return
+        await self.coordinator.client.async_delete(
+            self.entity_description.key, self.extra_state_attributes[ATTR_ID]
+        )
```

Comment:
need to check id to be deleted exists / log error

Reply:
You have a point here if there are no more entries for a specific endpoint. I will add a check to ensure that the sensor has an id in its attributes.